docs/configuration.md

```diff
@@ -201,10 +201,10 @@ If your input is ignored by `ignore` or `only`, Babel will print that this file
 Babel's configuration merging is relatively straightforward. Options will overwrite existing options
 when they are present and their value is not `undefined`. There are, however, a few special cases:
 
-- For `assumptions`, `parserOpts` and `generatorOpts`, objects are merged, rather than replaced, using the same logic as top-level options.
+- For `assumptions`, `parserOpts` and `generatorOpts`, objects are merged, rather than replaced.
 - For `plugins` and `presets`, they are replaced based on the identity of the plugin/preset object/function itself combined with the name of the entry.
 
-#### Option merging
+#### Option (except Plugin/Preset) merging
```

Comment:
Small thing: Do you consider Plugin and Preset proper nouns, hence why they are capitalized? I would have thought they should be lower-cased.

Reply:
Thanks for grammatical suggestions! They are copied from the next section.
```suggestion
#### Option (except plugin/preset) merging
```